dynamodb-stream-reader/src/main/java/elasticsearch/ElasticSearchIndexDocument.java

```diff
@@ -0,0 +1,109 @@
+package elasticsearch;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import nva.commons.utils.JsonUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ElasticSearchIndexDocument {
+
+    private static final Logger logger = LoggerFactory.getLogger(ElasticSearchIndexDocument.class);
+
+    private static final String CONTRIBUTORS_KEY = "contributor";
+    private static final String TITLE_KEY = "title";
+    private static final String DATE_KEY = "date";
+    private static final String RESOURCE_TYPE_KEY = "type";
+    public static final String ID_KEY = "id";
+    public static final String EMPTY_STRING = "";
+    private static final String NAME_KEY = "name";
+
+    private final String internalIdentifier;
+
+    private final String serviceUrl;
+    private final String indexName;
+    private final Map<String, Object> values = new HashMap<>();
+
+    /**
+     * Created an document suitable to upload to elasticsearch.
+     * @param indexName which elasticsearch index handles this document
+     * @param serviceUrl prefix of serviceurl for creating URI for resource source
+     * @param internalIdentifier shor internal identifier (uuid) for resource
+     */
+    public ElasticSearchIndexDocument(String indexName, String serviceUrl,  String internalIdentifier) {
+        this.indexName = indexName;
+        this.serviceUrl = serviceUrl;
+        this.internalIdentifier = internalIdentifier;
+        values.put(ID_KEY, serviceUrl + internalIdentifier);
+    }
+
+    public String getServiceUrl() {
+        return serviceUrl;
+    }
+
+    public String getIndexName() {
+        return indexName;
+    }
+
+    public String getInternalIdentifier() {
+        return internalIdentifier;
+    }
+
+    /**
+     * Adds a contributors name to the document.
+     * The document an hold multiple contributors
+     * @param name name of the contributor
+     */
+    public void addContributorName(String name) {
+        List<Object> contributors =
+                (List<Object>) values.computeIfAbsent(CONTRIBUTORS_KEY, l ->  new ArrayList<Object>());
```

Comment:
Isn't this just an initialization? It could be in the constructor in the form  

```
values.put(CONTRIBUTORS_KEY, new ArrayList<Object>())
```

Reply:
It could... moved to constructor